### PR TITLE
feat(repo-page): add functionality to see branch history

### PR DESCRIPTION
**Pull request by davidholyko**
#### Summary

Resolves: N/A
Card if applicable: [add functionality to lookup a specific branch history
](https://github.com/f1v/full-git-commit-history-app/projects/1#card-52608676)

- adds functionality to lookup branch commits

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in semantic format `TYPE(scope): description`

### CODE REVIEW

File: src/components/pages/RepoPage.js [hunk 1]
```diff
@@ -10,14 +10,32 @@ import { CommitList } from '../commit-list/CommitList';
...
+  let commits = [];
+
+  if (branch) {
+    commits = branchCommitHistory;
+  } else {
+    commits = (commitHistory[user] && commitHistory[user][repo]) || [];
+  }
+
+  /**
+   * picks up to full commit history for a repo's main branch
```
Comment (trybick) [inline src/components/pages/RepoPage.js]: typo "to" --> "the"

File: src/components/pages/RepoPage.js [hunk 2]
```diff
@@ -28,27 +46,46 @@ export const RepoPage = ({ match }) => {
...
+   * and saves in component state
+   */
   const getBranchesData = async () => {
     const rawData = await API.getRepoBranches({ owner: user, repo });
     const newBranches = rawData.map(({ name }) => name);
     setBranches(newBranches);
   };
 
+  /**
+   * returns to full commit history for selected repo branch
```
Comment (trybick) [inline src/components/pages/RepoPage.js]: same typo I think

File: src/components/routes/Routes.js [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { Route, Switch } from 'react-router-dom';
 import { UserPage, RepoPage, HomePage } from '../pages';
 
 export const Routes = () => {
+  // TODO: we can consolidate the render callback
```
Comment (trybick) [inline src/components/routes/Routes.js:6]: do you mean remove the `return` word below? should we do that now?

File: src/components/commit-list/CommitList.js [hunk 4]
```diff
@@ -37,31 +38,39 @@ const Commit = ({ githubCommitObject }) => {
...
-            </option>
-          );
-        })}
-      </Select>
-    </Flex>
-  );
+  const SelectDropdown = () => {
+    // TODO: add functionality for the edge case where a repo starts with main branch instead of master
+    return (
+      <Flex justifyContent="flex-end" mb="12px">
```
Comment (trybick) [inline src/components/commit-list/CommitList.js]: mb should be 25px unless you changed it

File: src/components/commit-list/CommitList.js [hunk 5]
```diff
@@ -37,31 +38,39 @@ const Commit = ({ githubCommitObject }) => {
...
+        <Select
+          borderColor="#808080"
+          color="#808080"
+          fontSize="15px"
+          onChange={onSelect}
+          w="160px"
```
Comment (trybick) [inline src/components/commit-list/CommitList.js:60]: We might need a value prop here. the Select always shows master as selected currently

**Comment by trybick** [comment]
Hey also when switching branches we should show a loader. in this PR or separate not sure tho.

File: src/components/routes/Routes.js (hunk 3 above)
Comment (davidholyko) [reply src/components/routes/Routes.js:6]: we keep the return. its hard to put comments in between parenthesis ( )

File: src/components/commit-list/CommitList.js (hunk 4 above)
Comment (davidholyko) [reply src/components/commit-list/CommitList.js]: seems to be a merge conflict error

**Comment by davidholyko** [comment]
> Hey also when switching branches we should show a loader. in this PR or separate not sure tho.

this will be in a separate bug. theres a memory leak bug that happens when trying to implement this right now

**Review by trybick — APPROVED** [review]
Nice job on the base64 conversion :D

**Pull request merged**